apps/asap-server/src/entities/event.ts

```diff
@@ -33,26 +30,58 @@ export const parseGraphQLEvent = (item: GraphqlEvent): EventResponse => {
     ? groups[0].thumbnail
     : undefined;
 
-  const meetingMaterials =
-    item.flatData?.meetingMaterials?.map(({ title, url, label }) => ({
+  const endDate = DateTime.fromISO(item.flatData!.endDate!);
+  const isStale = endDate.diffNow('days').get('days') < -10; // 10 days have passed after the event
+
+  const {
+    notesLocked,
+    videoRecordingLocked,
+    presentationLocked,
+    meetingMaterialsLocked,
+    notes,
+    videoRecording,
+    presentation,
+    meetingMaterials,
+  } = item.flatData!;
+
+  const notesRes =
+    notesLocked || (!notes && isStale)
```

Comment:
why don't use extract this logic into some other function and reuse it across all 4? I feel like it could be refactored as well. Something like...
```typescript
const getVal = (isPermUnavailable: boolean, val?: T): T | null | undefined => {
   if (isPermUnavailable === true) {
      return null;
   }

   if (!va; && isStale) {
  (....)
}

```